scripts/sentiment_analysis/process.py

```diff
@@ -0,0 +1,126 @@
+        output_size = 5
+        return train_dataset, test_dataset, output_size
+    elif data_name == 'SST-2':
+        train_dataset, test_dataset = [nlp.data.SST_2(root='data/sst-2', segment=segment)
+                                       for segment in ('train', 'test')]
+        output_size = 2
+        return train_dataset, test_dataset, output_size
+    elif data_name == 'Subj':
+        train_dataset = nlp.data.SUBJ(root='data/Subj', segment='all')
+        output_size = 2
+        return train_dataset, output_size
+    else:
+        train_dataset, test_dataset = [nlp.data.TREC(root='data/trec', segment=segment)
+                                       for segment in ('train', 'test')]
+        output_size = 6
+        return train_dataset, test_dataset, output_size
+
+def _clean_str(string, data_name):
+    if data_name == 'SST-1' or data_name == 'SST-2':
+        string = re.sub(r'[^A-Za-z0-9(),!?\'\`]', ' ', string)
+        string = re.sub(r'\s{2,}', ' ', string)
+        return string.strip().lower()
+    else:
+        string = re.sub(r'[^A-Za-z0-9(),!?\'\`]', ' ', string)
+        string = re.sub(r'\'s', ' \'s', string)
+        string = re.sub(r'\'ve', ' \'ve', string)
+        string = re.sub(r'n\'t', ' n\'t', string)
+        string = re.sub(r'\'re', ' \'re', string)
+        string = re.sub(r'\'d', ' \'d', string)
+        string = re.sub(r'\'ll', ' \'ll', string)
+        string = re.sub(r',', ' , ', string)
+        string = re.sub(r'!', ' ! ', string)
+        string = re.sub(r'\(', ' ( ', string)
+        string = re.sub(r'\)', ' ) ', string)
+        string = re.sub(r'\?', ' ? ', string)
+        string = re.sub(r'\s{2,}', ' ', string)
+        return string.strip() if data_name == 'TREC' else string.strip().lower()
+
+def _build_vocab(data_name, train_dataset, test_dataset):
+    all_token = []
+    max_len = 0
+    for i, line in enumerate(train_dataset):
+        train_dataset[i][0] = _clean_str(line[0], data_name)
+        line = train_dataset[i][0].split()
+        max_len = max_len if max_len > len(line) else len(line)
+        all_token.extend(line)
+    for i, line in enumerate(test_dataset):
+        test_dataset[i][0] = _clean_str(line[0], data_name)
+        line = test_dataset[i][0].split()
+        max_len = max_len if max_len > len(line) else len(line)
+    all_token.extend(line)
+    vocab = nlp.Vocab(nlp.data.count_tokens(all_token))
+    vocab.set_embedding(nlp.embedding.create('Word2Vec', source='GoogleNews-vectors-negative300'))
+    for line in vocab.embedding._idx_to_token:
+        if (vocab.embedding[line] == nd.zeros(300)).sum() == 300:
+            vocab.embedding[line] = nd.random.normal(-1.0, 1.0, 300)
+    vocab.embedding['<unk>'] = nd.zeros(300)
+    vocab.embedding['<pad>'] = nd.zeros(300)
+    vocab.embedding['<bos>'] = nd.zeros(300)
+    vocab.embedding['<eos>'] = nd.zeros(300)
+    print(max_len)
+    return vocab, max_len
+
+# Dataset preprocessing
+def preprocess(x, vocab, max_len):
```

Comment:
Add a one-line docstring to describe all public methods in scripts.
e.g.,
def preprocess():
"""Pre-process the data set.""" (Note that period . is required to complete a sentence.)